src/c_api/c_api.cc

```diff
@@ -149,6 +149,12 @@ struct XGBAPIThreadLocalEntry {
   std::vector<GradientPair> tmp_gpair;
 };
 
+XGB_DLL void XGBoostVersion(int* major, int* minor, int* patch) {
+  *major = XGBOOST_VER_MAJOR;
```

Comment:
Should we write to variable only if is not NULL?
I can envision this use case. We need to know only major and minor version to implement backward compatibility or something like that

then I would like to use this function as

int major;
int minor;
XGBoostVersion(major, minor, NULL);

What do you think?

This is minor comment

Reply:
@trams Good idea.